Packs/RetarusSecureEmailGateway/Integrations/RetarusSecureEmailGateway/RetarusSecureEmailGateway.py

```diff
@@ -0,0 +1,248 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+from CommonServerUserPython import *  # noqa
+import urllib3
+from websockets import Data
+from websockets.sync.client import connect
+from websockets.sync.connection import Connection
+from dateutil import tz
+import traceback
+import threading
+from contextlib import contextmanager
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+''' CONSTANTS '''
+VENDOR = "Retarus"
+PRODUCT = "Secure Email Gateway"
+FETCH_INTERVAL_IN_SECONDS = 60
+FETCH_SLEEP = 5
+SERVER_IDLE_TIMEOUT = 60
+DEFAULT_CHANNEL = "default"
+LOG_PREFIX = "Retarus-logs"
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+
+
+''' CLIENT CLASS '''
+class EventConnection:
+    def __init__(self, connection:Connection, fetch_interval: int = FETCH_INTERVAL_IN_SECONDS,
+                 idle_timeout: int = SERVER_IDLE_TIMEOUT):
+        self.connection = connection
+        self.lock = threading.Lock()
```

Comment:
Putting the lock here means we stop all other things from now, is it the latest we can use it?